app/actions/remote/entry.ts

```diff
@@ -21,18 +25,276 @@ import {MyPreferencesRequest, fetchMyPreferences} from './preference';
+        fetchMyChannelsForTeam(serverUrl, initialTeamId, includeDeletedChannels, lastDisconnected, fetchOnly),
+        fetchMyPreferences(serverUrl, fetchOnly),
+        fetchMe(serverUrl),
+    ];
+
+    const resolved = await Promise.all(promises);
+    const [teamData, chData, prefData, meData] = resolved; // eslint-disable-line @typescript-eslint/no-unused-vars
+
+    const removeTeamIds: string[] = [];
+
+    if (teamData.teams?.length === 0) {
+        // User is no longer a member of any team
+        const myTeams = await queryMyTeams(database);
+        myTeams?.forEach((myTeam) => removeTeamIds.push(myTeam.id), removeTeamIds);
+
+        return {
+            initialTeamId: '',
+            teamData,
+            chData,
+            prefData,
+            meData,
+            removeTeamIds,
+        };
+    }
+
+    const inTeam = teamData.teams?.find((t) => t.id === initialTeamId);
+    if (!inTeam || chData.error?.status_code === 403) {
+        // User is no longer a member of the current team
+        removeTeamIds.push(initialTeamId);
+
+        const availableTeamIds = await getAvailableTeamIds(database, teamData.teams, prefData.preferences, meData.user?.locale, initialTeamId);
+        const switchedTeamData = await switchTeams(serverUrl, availableTeamIds, removeTeamIds, includeDeletedChannels, lastDisconnected, fetchOnly);
+
+        return {
+            teamData,
+            chData,
+            prefData,
+            meData,
+            ...switchedTeamData,
+        };
+    }
+
+    return {initialTeamId, teamData, chData, prefData, meData, removeTeamIds};
+};
+
+const getAvailableTeamIds = async (database: Database, teams: Team[] | undefined, preferences: PreferenceType[] | undefined, locale: string | undefined, excludeTeamId: string): Promise<string[]> => {
+    let availableTeamIds: string[] = [];
+
+    if (teams) {
+        let teamOrderPreference;
+        if (preferences) {
+            teamOrderPreference = getPreferenceValue(preferences, Preferences.TEAMS_ORDER, '', '') as string;
+        } else {
+            const dbPreferences = await queryPreferencesByCategoryAndName(database, Preferences.TEAMS_ORDER, '');
+            teamOrderPreference = dbPreferences[0].value;
+        }
+
+        const userLocale = locale || (await queryCurrentUser(database))?.locale;
+        const {config} = await queryCommonSystemValues(database);
+
+        const defaultTeam = selectDefaultTeam(teams, userLocale, teamOrderPreference, config.ExperimentalPrimaryTeam);
+
+        availableTeamIds = [defaultTeam!.id];
+    } else {
+        const dbTeams = await queryMyTeams(database);
+        if (dbTeams) {
+            availableTeamIds = dbTeams.map((team) => team.id);
+        }
+    }
+
+    return availableTeamIds.filter((id) => id !== excludeTeamId);
+};
+
+const switchTeams = async (serverUrl: string, availableTeamIds: string[], removeTeamIds: string[], includeDeleted = true, since = 0, fetchOnly = false) => {
```

Comment:
perhaps change the name so is not confused with an action to actually switch Teams ?